src/lib/y2firewall/proposal_settings.rb

```diff
@@ -0,0 +1,106 @@
+require "yast"
+
+module Y2Firewall
+  class ProposalSettings
+    include Yast::Logger
+    include Yast::I18n
+    attr_accessor :enable_firewall, :enable_sshd, :open_ssh, :open_vnc
+
+    def initialize
+      Yast.import "PackagesProposal"
+      Yast.import "ProductFeatures"
+      Yast.import "Linuxrc"
+
+      load_features
+      enable_firewall! if @enable_firewall
+      enable_vnc! if Yast::Linuxrc.vnc
+      enable_sshd! if Yast::Linuxrc.usessh || @enable_sshd
+      open_ssh! if Yast::Linuxrc.usessh || @open_ssh
+    end
+
+    def load_features
+      load_feature(:enable_firewall, :enable_firewall)
+      load_feature(:firewall_enable_ssh, :open_ssh)
+      load_feature(:enable_sshd, :enable_sshd)
+    end
+
+    # Services
+
+    def enable_firewall!
+      Yast::PackagesProposal.AddResolvables("firewall", :package, ["firewalld"])
+
+      log.info "Enabling Firewall"
+      self.enable_firewall = true
+    end
+
+    def disable_firewall!
+      Yast::PackagesProposal.RemoveResolvables("firewall", :package, ["firewalld"])
```

Comment:
Do you want to uninstall the package? I am not sure if the old code did that.

Reply:
Same behavior than before:

https://github.com/yast/yast-network/blob/master/src/clients/firewall_stage1_proposal.rb#L404

Reply:
Oh, I see, the stage1 proposal.